### PR TITLE
Creating tar only if the backup name is valid (bsc#966424)

### PR DESCRIPTION
**Pull request by rsalevsky**
- create archive only if the name is valid
- run custom validations only in upload case
- don't validate version and size

### CODE REVIEW

File: crowbar_framework/app/models/backup.rb [hunk 1]
```diff
@@ -120,12 +116,10 @@ def attribute_names
...
   protected
 
-  def save_or_create_archive
-    if name.blank?
-      save_archive
-    else
-      create_archive
-    end
+  def upload?
+    return false unless file.present?
```
Comment (tboerger) [inline crowbar_framework/app/models/backup.rb]: for this cases i generally prefer

```
if file.present?
  true
else
  false
end
```

**Comment by tboerger** [comment]
Some NITPICK otherwise :+1:

**Comment by MaximilianMeister** [comment]
+1

File: crowbar_framework/app/models/backup.rb [hunk 2]
```diff
@@ -32,16 +33,11 @@ class Backup < ActiveRecord::Base
       message: "allows only letters and numbers"
     }
 
-  validates :version,
-    presence: true
-
-  validates :size,
-    presence: true
```
Comment (vuntz) [inline crowbar_framework/app/models/backup.rb]: I don't understand why this gets removed. Can this  be explained in the commit message?

Comment (MaximilianMeister) [inline crowbar_framework/app/models/backup.rb]: agree that the commit message should explain it. it is not validated, as the values are set by crowbar, so we assume they are valid.

**Comment by rhafer** [comment]
+1

**Comment by vuntz** [comment]
+1, can you backport to 3.0?

**Pull request merged**